tests/beta/test_beta_qnode.py

```diff
@@ -0,0 +1,900 @@
+        assert qn.qtape is not old_tape
+
+    def test_jacobian(self, tol):
+        """Test the jacobian computation"""
+        dev = qml.device("default.qubit", wires=2)
+
+        def func(x, y):
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return qml.probs(wires=0), qml.probs(wires=1)
+
+        qn = QNode(func, dev, diff_method="finite-diff", h=1e-8, approx_order=2)
+        assert qn.gradient_kwargs["h"] == 1e-8
+        assert qn.gradient_kwargs["approx_order"] == 2
+
+        jac = qn.gradient_fn(qn)(0.45, 0.1)
+        assert jac.shape == (2, 2, 2)
+
+    def test_returning_non_measurements(self):
+        """Test that an exception is raised if a non-measurement
+        is returned from the QNode."""
+        dev = qml.device("default.qubit", wires=2)
+
+        def func(x, y):
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return 5
+
+        qn = QNode(func, dev)
+
+        with pytest.raises(
+            qml.QuantumFunctionError, match="must return either a single measurement"
+        ):
+            qn(5, 1)
+
+        def func(x, y):
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0)), 5
+
+        qn = QNode(func, dev)
+
+        with pytest.raises(
+            qml.QuantumFunctionError, match="must return either a single measurement"
+        ):
+            qn(5, 1)
+
+    def test_inconsistent_measurement_order(self):
+        """Test that an exception is raised if measurements are returned in an
+        order different to how they were queued on the tape"""
+        dev = qml.device("default.qubit", wires=2)
+
+        def func(x, y):
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            m = qml.expval(qml.PauliZ(0))
+            return qml.expval(qml.PauliX(1)), m
+
+        qn = QNode(func, dev)
+
+        with pytest.raises(
+            qml.QuantumFunctionError,
+            match="measurements must be returned in the order they are measured",
+        ):
+            qn(5, 1)
+
+    def test_consistent_measurement_order(self):
+        """Test evaluation proceeds as expected if measurements are returned in the
+        same order to how they were queued on the tape"""
+        dev = qml.device("default.qubit", wires=2)
+
+        def func(x, y):
+            global op1, op2, op3, m1, m2
+            op1 = qml.RX(x, wires=0)
+            op2 = qml.RY(y, wires=1)
+            op3 = qml.CNOT(wires=[0, 1])
+            m1 = qml.expval(qml.PauliZ(0))
+            m2 = qml.expval(qml.PauliX(1))
+            return [m1, m2]
+
+        qn = QNode(func, dev)
+        qn(5, 1)  # evaluate the QNode
+        assert qn.qtape.operations == [op1, op2, op3]
+        assert qn.qtape.measurements == [m1, m2]
+
+    @pytest.mark.xfail

```

Comment:
I was just going to point out that it was exciting that we now take into account qubit-wise commutativity with the measured observables, but then I saw this is marked as `xfail`, why won't this pass, is it just not implemented in this version of the QNode yet?

Reply:
It is implemented in #1651 🙂

Reply:
Hooray!! :tada: